news/_posts/starter-theme/2015-12-17-starter-theme-news-3.markdown

```diff
@@ -0,0 +1,97 @@
+
+The Starter Theme also displays a cart summary, without any actions possible, on the right. It allows the customer to always know what he is buying, how much it will cost him, etc.
+
+![PrestaShop New Checkout Process](/assets/images/2015/12/new_checkout_process.png)
+
+## Payment API
+
+To provide the best user experience possible for customers, and to guarantee that PrestaShop orders respect the law in all countries, we implemented a new payment API. To do that, we chose to respect the strictest requirements we know about : the German ones.
+For example, the Terms & Conditions checkbox is now displayed with the payment options, and there is only one button to pay.
+
+That way, we have more control of the layout of payment options.
+
+To help you to create or adapt your payment modules, we have reworked bankwire and developed **paymentexample**. This module is an example of what you can do. It implements 4 types of payment modules, working:
+
+* offline: call a simple URL, like bankwire or cheque
+
+* externally: call a simple URL but a third-party webserver processes the payment, like PayPal
+
+* embedded: write the credit card details on the website directly in a form, like Stripe
+
+* in iframe: payment form displayed inside an iframe, like Atos
+
+{% highlight php %}
+// PHP
+public function hookPaymentOptions($params)
+{
+  if (!$this->active) {
+      return;
+  }
+
+  if (!$this->checkCurrency($params['cart'])) {
+      return;
+  }
+
+  $this->getTemplateVarInfos();
+
+  $newOption = new PaymentOption();
+  $newOption->setCallToActionText($this->l('Pay by Bank Wire'))
+                ->setAction($this->context->link->getModuleLink($this->name, 'validation', array(), true))
+                ->setAdditionalInformation($this->context->smarty->fetch(implode(DIRECTORY_SEPARATOR, [__DIR__, 'views', 'templates', 'front', 'payment_execution.tpl'])))
+                ->setLogo(Media::getMediaPath(_PS_MODULE_DIR_.$this->name.'/bankwire.jpg'));
+
+  $payment_options = [
+      $newOption,
+  ];
+
+  return $payment_options;
+}
+
+{% endhighlight %}
+
+## Faceted/layered navigation, pagination and search
+
+In order to simplify the `.tpl` and `.js` files for all the faceted navigation, pagination and search, and to remove all the calculation from the front office (which is one of the main purpose of the StarterTheme) we have fully rewritten the whole thing. @djfm was responsible for that part. It would be too complicated to easily explain or transcript the work here, so don't hesitate to look at it on [GitHub](https://github.com/PrestaShop/PrestaShop/pull/4548/files).
+
+Accordind to standards, we have chosen to call “Filter” a value which helps us to find the corresponding products (like red, XL or a price range) and to call “Facet” a collection of filters (like color, size or price). From now on we'll also refer to “Faceted navigation” when in previous version we were mentioning layered navigation.
+
```

Comment:
+ one of the main innovations in this part of the software is that now all product listing controllers (Manufacturer, Supplier, Search...) share the same behaviour and can all delegate the search to modules in a nice, standard fashion.